backend/corpora/common/corpora_orm.py

```diff
@@ -14,18 +15,57 @@
     String,
     text,
 )
+from sqlalchemy.ext.declarative import DeclarativeMeta
 from sqlalchemy.ext.declarative import declarative_base
 from sqlalchemy.orm import relationship, sessionmaker
 
 pkg_root = os.path.abspath(os.path.join(os.path.dirname(__file__), ".."))  # noqa
 sys.path.insert(0, pkg_root)  # noqa
 
 from common.corpora_config import CorporaDbConfig
+from common.utils.json import CustomJSONEncoder
 
 
-Base = declarative_base()
-deployment_stage = os.environ["DEPLOYMENT_STAGE"]
+class TransformingBase(object):
```

Comment:
I'm curious about this class as a whole. I don't see `to_dict` or `to_json` being used anywhere -- what's the use case you're thinking of here and why specifically convert the database object to dictionary and JSON instead of the application layer objects?

Also what's the benefit of having a class in the ORM layer versus a utility function that could perform the transformation?

Reply:
The primary use case is when a user make an api request they will want a json response. 

The benefit of having it in the ORM layer is that we only need one conversion function for all of the DB objects. It also allows us to follow the relationships defined by the ORM and convert those object as well in a single function.